ntl.c

```diff
@@ -326,6 +326,8 @@ STATIC ntl_t ntl_append(ntl_t p, size_t elem_size, void *added_elem)
     i++;
   }
   memcpy(o[i], added_elem, elem_size);
+  // elem inner-fields are now owned by NTL
+  memset(added_elem, 0, elem_size);
   return o;
```

Comment:
This is a tricky one.  I would avoid introducing any ownership concept into the code.    once it's added, it opens a can of warms.

Reply:
I see, the problem is that the NTL will implicitly take ownership. For example:
```c
struct discord_channel *channel = malloc(sizeof(struct discord_channel));
discord_channel_init(channel);
 
 NTL_T(struct discord_channel) channels = NULL;
ntl_append2((ntl_t*)&channels, sizeof(struct discord_channel), channel);

discord_channel_list_free(channels); // this will free any `channel` inner structs
discord_channel_cleanup(channel); // this will lead to a double-free if those inner structs have been freed
free(channel);
```

Reply:
I added a new commit BTW: 0cc1d04

Reply:
Will the following code work? 
```
struct discord_channel *channel = malloc(sizeof(struct discord_channel));
discord_channel_init(channel);
 
NTL_T(struct discord_channel) channels = NULL;
ntl_append2((ntl_t*)&channels, sizeof(struct discord_channel), channel);

discord_channel_list_free(channels); 
free(channel);
```

Reply:
It might still occur a double-free, as `channels` will create a copy of `channel`, including the addresses it might be pointing to ex `channel->messages` will be accessible with `channels[0]->messages`.

Reply:
Let's push the responsibility of correctly using this function to the users.   This is consistent with the principle of C.

Reply:
Understood, I'll do a rollback then. How is the latest commit?